cdk/lib/stack.ts

```diff
@@ -1,32 +1,31 @@
-import { Vpc, SecurityGroup, Peer, Port } from '@aws-cdk/aws-ec2';
-import { Cluster } from '@aws-cdk/aws-ecs';
-import { Stack, Construct, StackProps } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { aws_ec2 as ec2, aws_ecs as ecs, Stack, StackProps } from 'aws-cdk-lib';
 
 import { CacclAppEnvironment } from './appEnvironment';
+import { CacclSshBastion } from './bastion';
+import { CacclCache, CacclCacheOptions } from './cache';
 import { CacclMonitoring } from './dashboard';
-import { CacclDbOptions, CacclDbBase } from './db';
+import { CacclDbOptions, createDbConstruct } from './db';
 import { CacclLoadBalancer, LoadBalancerSecurityGoups } from './lb';
 import { CacclNotifications, CacclNotificationsProps } from './notify';
-import { CacclCache, CacclCacheOptions } from './cache';
+import { CacclScheduledTask, CacclScheduledTasks } from './scheduledTasks';
 import { CacclService } from './service';
 import { CacclTaskDef, CacclTaskDefProps } from './taskdef';
-import { CacclSshBastion } from './bastion';
-import { CacclScheduledTask, CacclScheduledTasks } from './scheduledTasks';
 
 export interface CacclDeployStackProps extends StackProps {
   vpcId?: string;
   cidrBlock?: string;
   maxAzs: number;
   certificateArn: string;
   ecsClusterName?: string;
-  appEnvironment: { [key: string]: string; };
+  appEnvironment: { [key: string]: string };
```

Comment:
Seems like the environment type is a shared type `{ [k: string]: string }` so maybe it's useful to just create a shared type in a file somewhere so you can reuse it? `appEnvironment: AppEnvironment`